src/commands/branch/merge.ts

```diff
@@ -0,0 +1,75 @@
+import { Flags } from "@oclif/core";
+
+import BaseCommand from "@/lib/base-command";
+import { CustomArgs } from "@/lib/helpers/arg";
+import { KnockEnv } from "@/lib/helpers/const";
+import { withSpinnerV2 } from "@/lib/helpers/request";
+import { promptToConfirm } from "@/lib/helpers/ux";
+
+export default class BranchMerge extends BaseCommand<typeof BranchMerge> {
+  static summary = "Merges a branch into the development environment.";
+
+  static args = {
+    slug: CustomArgs.slug({
+      required: true,
+      description: "The slug of the branch to merge",
+    }),
+  };
+
+  static flags = {
+    force: Flags.boolean({
+      summary: "Remove the confirmation prompt.",
+    }),
+    "skip-deletion": Flags.boolean({
+      summary: "Skip deleting the branch after merging.",
+    }),
+  };
+
+  async run(): Promise<void> {
+    const { args, flags } = this.props;
+
+    const promotePrompt = `Merge all changes from branch \`${args.slug}\` into the development environment?`;
+    const shouldPromote = flags.force || (await promptToConfirm(promotePrompt));
+    if (!shouldPromote) return;
+
+    await this.promoteBranchCommits();
+
+    const deletePrompt = `Delete branch \`${args.slug}\`?`;
+    const shouldDeleteBranch =
+      !flags["skip-deletion"] &&
+      (flags.force || (await promptToConfirm(deletePrompt)));
+
+    if (shouldDeleteBranch) {
+      await this.deleteBranch();
+    }
```

Comment:
```suggestion
    if (flags["skip-deletion"]) return;

    const deletePrompt = `Delete branch \`${args.slug}\`?`;
    const shouldDeleteBranch = (flags.force || (await promptToConfirm(deletePrompt)));
    if (!shouldDeleteBranch) return;
    
    await this.deleteBranch();
```
Nit, I'd break out this logic into individual parts and early returns, to make it easier to read and use the similar patterns.